content/angular-authentication/index.md

```diff
@@ -0,0 +1,1460 @@
+---
+title: Angular Authentication
+description: "Learn how Angular authentication works, key methods to secure apps, and how to integrate with providers like SuperTokens or OAuth."
+date: "2025-05-18"
+cover: "TODO.png"
+category: "programming, angular, authentication"
+author: "Maurice Saldivar"
+---
+
+# Angular Authentication – A Complete Guide for Secure Web Apps
+
+## What Is Authentication in Angular?
+
+Before a user can access protected components, routes, or APIs Angular must first Authenticate them. This verification is fundamental to web security, creating a barrier that prevents unauthorized users from accessing sensitive information or performing restricted actions. 
+
+## Authentication From A High Level
+
+Authentication is essentially about confirming “who you are” before granting access. In Angular applications, this involves validating users, like usernames and passwords, or verifying tokens from a trusted identity provider. When implemented properly, authentication creates a secure foundation for your application by ensuring only verified users can enter protected areas. 
+
+## Why It Matters
+
+* **Route Protection:** Prevents unauthorized access to sensitive parts of your application  
+* **Data Security:** Safeguards account information  
+* **API Protection:** Ensures only authenticated users can make requests to specific backend services   
+* **Regulatory Compliance:** Many industries require proper authentication for legal standard, e.g. SOC 2   
+*  **User Trust:** A secure systems builds confidence in your platform
+
+Without robust authentication, your Angular application becomes vulnerable to countless security threats, it's only a matter of time until user data is compromised. 
+
+## Angular’s Role
+
+Angular itself doesn’t implement authentication protocols directly. Instead, it provides a framework that makes integration with authentication systems a simple reality. 
+
+* **Client-side Routing:** Angular’s router enables protection of routes through guards that can prevent navigation to restricted routes.  
+* **HTTP Interceptors:**  Automatically attach authorization tokens, handle token refreshes, and manage authentication errors.  
+* **Reactive Forms:** Handle secure collection of user credentials.   
+* **Services Pattern:** Provides a clean battle tested architecture for authentication. 
+
+Remember for the actual identity verification, Angular relies on external authentication providers. These can range from your own custom backend solution to open-source solutions like SuperTokens. 
+
+The separation of concerns allows Angular applications to integrate with virtually any authentication system while maintaining a clean architecture focused on delivering the core user experience.  
+
+## Authentication vs Authorization in Angular
```

Comment:
Link to the authentication vs authorization post for readers who want to know more